spring-webflux/src/main/java/org/springframework/web/reactive/filter/HiddenHttpMethodFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.web.reactive.filter;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.util.Assert;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+
+import java.util.Locale;
+
+/**
+ * Reactive {@link WebFilter} that converts posted method parameters into HTTP methods,
+ * retrievable via {@link ServerHttpRequest#getMethod()}. Since browsers currently only
+ * support GET and POST, a common technique - used by the Prototype library, for instance -
+ * is to use a normal POST with an additional hidden form field ({@code _method})
+ * to pass the "real" HTTP method along. This filter reads that parameter and changes
+ * the {@link ServerHttpRequest#getMethod()} return value using {@link ServerWebExchange#mutate()}.
+ *
+ * <p>The name of the request parameter defaults to {@code _method}, but can be
+ * adapted via the {@link #setMethodParam(String) methodParam} property.
+ *
+ * <p><b>NOTE: This filter needs to run after multipart processing in case of a multipart
```

Comment:
I don't think this note applies here, since we don't have such a filter and request multipart support is not done yet (see https://jira.spring.io/browse/SPR-14546). Also, reactive web applications are only running with embedded web servers, so `web.xml` does not apply here.